kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java

```diff
@@ -251,38 +266,52 @@ public void testGetSolvers() {
         assertNull(returnedInstance.getScoreWrapper().toScore());
     }
 
-    @Test
+    @Test(timeout = 15_000)
     public void testExecuteSolver() throws Exception {
+        testExecuteSolverHelper(SOLVER_1_ID, false);
+    }
+
+    @Test(timeout = 15_000)
+    public void testPrometheusMetrics() throws Exception {
+        testExecuteSolverHelper(SOLVER_PROMETHEUS_ID, true);
+    }
+
+    private void testExecuteSolverHelper(String solverID, boolean checkPrometheusMetrics) throws Exception {
```

Comment:
Just Nitpicking: instead of passing boolean flag, you can call the `checkPrometheusMetrics` method from the end of `testPrometheusMetrics` and skip it in the `testExecuteSolver`.